### DISCUSSION TITLE
Desktop Icon Config

### DISCUSSION THREAD

**Discussion by bomshteyn** [NativePHP/laravel#201] [Q&A] [answered]
Is  there a way to replace the default NativePHP icon when publishing an app?

I looked through the configuration docs here: https://nativephp.com/docs/1/getting-started/configuration but don't see anyway to change the main icon.

The only reference to changing an icon i see is on the menubar: https://nativephp.com/docs/1/the-basics/menu-bar#menu-bar-icon

**Comment by hussainweb** [comment]
I couldn't find any configuration to do this but I got it working by overwriting the icon files in the package.
```json
        "post-install-cmd": [
            // ...
            "cp resources/icons/Icon.png ./vendor/nativephp/electron/resources/js/resources/icon.png",
            "cp resources/icons/Icon.png ./vendor/nativephp/electron/resources/js/build/icon.png"
        ],
        "post-update-cmd": [
            // ...
            "cp resources/icons/Icon.png ./vendor/nativephp/electron/resources/js/resources/icon.png",
            "cp resources/icons/Icon.png ./vendor/nativephp/electron/resources/js/build/icon.png"
        ],
```

Of course, you need to create the `resources/icons` directory and place the `Icon.png` file in there.

**Reply by eugenefvdm** [reply]
Thanks @hussainweb this worked for me on Mac Apple Silicon.

**Reply by theoreaves** [reply]
Forgive me.  I'm new to this type of development.  What file is that?

**Reply by alnutile** [reply]
@theoreaves this is the composer.json file in the root of the project.

this area
```json
    "scripts": {
        "post-autoload-dump": [
            "Illuminate\\Foundation\\ComposerScripts::postAutoloadDump",
            "@php artisan package:discover --ansi"
        ],
        "post-update-cmd": [
            "@php artisan vendor:publish --tag=laravel-assets --ansi --force"
        ],
        "post-root-package-install": [
            "@php -r \"file_exists('.env') || copy('.env.example', '.env');\""
        ],
        "post-create-project-cmd": [
            "@php artisan key:generate --ansi",
            "@php -r \"file_exists('database/database.sqlite') || touch('database/database.sqlite');\"",
            "@php artisan migrate --graceful --ansi"
        ],
        "serve": "php artisan native:serve",
        "fix": "vendor/bin/pint",
        "test": "php artisan test",
        "stan": "vendor/bin/phpstan analyse --memory-limit 2G"
    },
```

**Reply by peskybogle** [reply]
Just adding a comment here for others, as I couldn't work out why this wasn't working for me with a test icon.

It requires at least a 512px x 512px image. As soon as I used a larger png it started working fine.

**Comment by simonhamp** [comment, marked as answer]
A quick update, I opened [a PR for this](https://github.com/NativePHP/electron/pull/103) which is a much smoother approach. Closing this thread as this will be merged soon.
